test/y2partitioner/dialogs/lvm_lv_size_test.rb

```diff
@@ -0,0 +1,222 @@
+        let(:value) { :max_size }
+
+        let(:max_size) { 10.GiB }
+
+        it "sets #size in the controller to max possible size" do
+          widget.store
+          expect(controller.size).to eq(max_size)
+        end
+      end
+
+      context "when custom size is selected" do
+        let(:value) { :custom_size }
+
+        let(:custom_size) { 1.GiB }
+
+        it "sets #size in the controller to the value indicated" do
+          widget.store
+          expect(controller.size).to eq(custom_size)
+        end
+      end
+    end
+  end
+
+  describe Y2Partitioner::Dialogs::LvmLvSize::MaxSizeDummy do
+    subject(:widget) { described_class.new(1.GiB) }
+
+    include_examples "CWM::AbstractWidget"
+  end
+
+  describe Y2Partitioner::Dialogs::LvmLvSize::CustomSizeInput do
+    subject(:widget) { described_class.new(5.GiB, 1.GiB, 10.GiB) }
+
+    before do
+      allow(widget).to receive(:value).and_return(value)
+    end
+
+    let(:value) { nil }
+
+    include_examples "CWM::AbstractWidget"
+
+    describe "#validate" do
+      context "when indicated value in not a valid size" do
+        let(:value) { nil }
+
+        it "returns false" do
+          expect(widget.validate).to eq(false)
+        end
+
+        it "shows an error popup" do
+          expect(Yast::Popup).to receive(:Error)
+          widget.validate
+        end
+      end
+
+      context "when indicated value in less than min size" do
+        let(:value) { 1.KiB }
+
+        it "returns false" do
+          expect(widget.validate).to eq(false)
+        end
+
+        it "shows an error popup" do
+          expect(Yast::Popup).to receive(:Error)
+          widget.validate
+        end
+      end
+
+      context "when indicated value in bigger than max size" do
+        let(:value) { 100.GiB }
+
+        it "returns false" do
+          expect(widget.validate).to eq(false)
+        end
+
+        it "shows an error popup" do
+          expect(Yast::Popup).to receive(:Error)
+          widget.validate
+        end
+      end
+
+      context "when indicated value in between min and max sizes" do
+        let(:value) { 6.GiB }
+
+        it "returns true" do
+          expect(widget.validate).to eq(true)
+        end
+
+        it "does not show an error popup" do
+          expect(Yast::Popup).to_not receive(:Error)
+          widget.validate
+        end
+      end
+    end
+  end
+
+  describe Y2Partitioner::Dialogs::LvmLvSize::StripesWidget do
+    subject(:widget) { described_class.new(controller) }
+
+    include_examples "CWM::CustomWidget"
+
+    describe "#store" do
+      before do
+        allow(Y2Partitioner::Dialogs::LvmLvSize::StripesNumberSelector)
+          .to receive(:new).and_return(stripes_number_widget)
+
+        allow(Y2Partitioner::Dialogs::LvmLvSize::StripesSizeSelector)
+          .to receive(:new).and_return(stripes_size_widget)
+      end
+
+      let(:stripes_number_widget) { double("StripesNumberSelector", value: stripes_number) }
```

Comment:
I tend to use `instance_double` when possible because it checks if stubbed methods exist in the original class and I think it worth it.